<launch>
  <node pkg="rosbag" type="play" name="player" output="log" args="--clock /home/robin/ros_orb_ws/rosbags/V2_03_difficult.bag"/>

  <node pkg="tf" type="static_transform_publisher" name="broadcaster" args="0 0 0 0 0 0 1 world map 100" /> 

  <node name="orb_slam3_mono" pkg="orb_slam3_ros"
      type="orb_slam3_ros_mono" output="screen">

       <remap from="/camera/image_raw" to="/cam0/image_raw"/>  

       <param name="publish_pointcloud" type="bool" value="true" />
       <param name="publish_pose" type="bool" value="true" />
       <param name="localize_only" type="bool" value="false" />
       <param name="reset_map" type="bool" value="false" />

       <!-- static parameters -->
       <param name="load_map" type="bool" value="false" />
       <param name="map_file" type="string" value="map.bin" />
       <!--param name="settings_file" type="string" value="/home/robin/ros_orb_ws/src/orb_slam_2_ros/orb_slam2/config/vi_euroc.yaml" /-->
       <param name="voc_file" type="string" value="/home/robin/orb3_ros_ws/src/orb_slam3_ros/orb_slam3/Vocabulary/ORBvoc.txt" />

       <param name="pointcloud_frame_id" type="string" value="map" />
       <param name="camera_frame_id" type="string" value="camera_link" />
       <param name="min_num_kf_in_map" type="int" value="5" />

       <!-- ORB parameters -->
       <param name="/ORBextractor/nFeatures" type="int" value="1000" />
       <param name="/ORBextractor/scaleFactor" type="double" value="1.2" />
       <param name="/ORBextractor/nLevels" type="int" value="8" />
       <param name="/ORBextractor/iniThFAST" type="int" value="20" />
       <param name="/ORBextractor/minThFAST" type="int" value="7" />

       <!-- Camera parameters -->
       <!-- Camera frames per second -->
       <param name="camera_fps" type="int" value="20" />
       <!-- Color order of the images (0: BGR, 1: RGB. It is ignored if images are grayscale) -->
       <param name="camera_rgb_encoding" type="bool" value="true" />

        <!-- Camera calibration parameters -->
       <param name="load_calibration_from_cam" type="bool" value="false" />
       <!-- Camera calibration and distortion parameters (OpenCV) -->
      <param name="camera_fx" type="double" value="458.654" />
      <param name="camera_fy" type="double" value="457.296" />
      <param name="camera_cx" type="double" value="367.215" />
      <param name="camera_cy" type="double" value="248.375" />

       <!-- Camera calibration and distortion parameters (OpenCV) -->
      <param name="camera_k1" type="double" value="-0.28340811" />
      <param name="camera_k2" type="double" value="0.07395907" />
      <param name="camera_p1" type="double" value="0.00019359" />
      <param name="camera_p2" type="double" value="1.76187114e-05" />
      <param name="camera_k3" type="double" value="0.0" />

      <!--Close/far Threshold. Baseline times.-->
      <param name="ThDepth" type="double" value="40" />
      <param name="DepthMapFactor" type="double" value="1.0" />
  </node>

  <node name="orb_slam3_server" pkg="orb_slam3_ros" type="orb_slam3_ros_server" output="screen" >
    <param name="voc_file" type="string" value="/home/robin/orb3_ros_ws/src/orb_slam3_ros/orb_slam3/Vocabulary/ORBvoc.txt" />
  </node>
  
  <node type="rviz" name="rviz" pkg="rviz" args="-d /home/robin/ros_orb_ws/src/orb_slam_2_ros/ros/config/rviz_config.rviz"/>
</launch>